src/UglyToad.PdfPig.Fonts/TrueType/Tables/CMapSubTables/Format4CMapTable.cs

```diff
@@ -56,6 +57,58 @@ public int CharacterCodeToGlyphIndex(int characterCode)
             return 0;
         }
 
+        public int[] GetCharactersCode()
+        {
+            var charactersCode = new List<int>();
+
+            ushort minCharacterStart = 0xffff;
+
+            foreach (var segmentIndex in Enumerable.Range(0, Segments.Count - 1))
+            {
+                var segment = Segments[segmentIndex];
+
+                minCharacterStart = Math.Min(minCharacterStart, (ushort)segment.StartCode);
+
+                var characterCountInSegment = (segment.EndCode - segment.StartCode) + 1;
+                if (segment.IdRangeOffset == 0)
+                {
+                    var chars = Enumerable.Range(segment.StartCode, characterCountInSegment);
+                    charactersCode.AddRange(chars);
+                }
+                else
+                {
+                    var chars = Enumerable.Range(0, characterCountInSegment).Where(i =>
+                    {
+                        var offset = segment.IdRangeOffset / 2 + i;
+                        var charCodeIndex = offset - Segments.Count + segmentIndex;
+
+                        return charCodeIndex >= 0 && charCodeIndex < GlyphIds.Count;
+                    }).Select(c => c + segment.StartCode);
+                    charactersCode.AddRange(chars);
+                }
+            }
+
+            if (minCharacterStart < 0xf000)
+            {
+                return charactersCode.ToArray();
+            }
+
+            if (minCharacterStart < 0xf100)
```

Comment:
I gathered this masks from [this](https://github.com/UglyToad/PdfPig/blob/dc55fd80113f8145b4f00f4afdb7fa808dc4a9fa/src/UglyToad.PdfPig.Fonts/TrueType/Tables/CMapTable.cs#L72-L96), but I don't know if there is a pattern that I should follow